instrumentation/message-handler/library/src/main/java/io/opentelemetry/instrumentation/messagehandler/BatchMessageHandler.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.messagehandler;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessageOperation;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+public abstract class BatchMessageHandler<T> {
+  protected String messagingOperation;
+  protected OpenTelemetry openTelemetry;
+  protected SpanNameExtractor<Collection<T>> spanNameExtractor;
+
+  protected Instrumenter<Collection<T>, Void> messageInstrumenter;
+
+  public BatchMessageHandler(OpenTelemetry openTelemetry) {
+    this(openTelemetry, MessageOperation.RECEIVE.name());
+  }
+
+  public BatchMessageHandler(OpenTelemetry openTelemetry, String messageOperation) {
+    this(openTelemetry, messageOperation, messages -> "Batch Message");
```

Comment:
I think there is no "Batch Message" but a batch of messages, or a "Message batch". Please check the messaging semantic conventions for what might be a reasonable default name, or maybe we should not provide one (I think "<destination name> process" should be what is the guideline, but that needs a way to get the destination (aka queue) name).